nullaway/src/main/java/com/uber/nullaway/fixserialization/location/MethodParameterLocation.java

```diff
@@ -30,22 +30,24 @@
 public class MethodParameterLocation extends AbstractFixLocation {
 
   /** Symbol of the targeted method. */
-  protected final Symbol.MethodSymbol enclosingMethod;
+  private final Symbol.MethodSymbol enclosingMethod;
   /** Symbol of the targeted method parameter. */
-  protected final Symbol.VarSymbol variableSymbol;
+  private final Symbol.VarSymbol variableSymbol;
   /** Index of the method parameter in the containing method's argument list. */
-  protected final int index;
+  private final int index;
 
   public MethodParameterLocation(Symbol target) {
     super(ElementKind.PARAMETER, target);
     this.variableSymbol = (Symbol.VarSymbol) target;
-    Symbol enclosingMethod = target;
+    Symbol cursor = target;
     // Look for the enclosing method.
-    while (enclosingMethod != null && enclosingMethod.getKind() != ElementKind.METHOD) {
-      enclosingMethod = enclosingMethod.owner;
+    while (cursor != null
+        && (cursor.getKind() != ElementKind.CONSTRUCTOR
+            && cursor.getKind() != ElementKind.METHOD)) {
```

Comment:
No need for nesting if all conditions are being joined by `&&`, the following would work just as well:

```
while(cursor != null 
      && cursor.getKind() != ElementKind.CONSTRUCTOR 
      && cursor.getKind() != ElementKind.METHOD) {
...
}
```